pkg/util/ovs/fake_ovs.go

```diff
@@ -0,0 +1,447 @@
+		}
+		if err := checkUnsupportedField(flow, parsed, "out_group"); err != nil {
+			return nil, err
+		}
+	}
+
+	if (fieldSet(parsed, "nw_src") || fieldSet(parsed, "nw_dst")) &&
+		!(fieldSet(parsed, "ip") || fieldSet(parsed, "arp") || fieldSet(parsed, "tcp") || fieldSet(parsed, "udp")) {
+		return nil, fmt.Errorf("bad flow %q (specified nw_src/nw_dst without ip/arp/tcp/udp)", flow)
+	}
+	if (fieldSet(parsed, "arp_spa") || fieldSet(parsed, "arp_tpa") || fieldSet(parsed, "arp_sha") || fieldSet(parsed, "arp_tha")) && !fieldSet(parsed, "arp") {
+		return nil, fmt.Errorf("bad flow %q (specified arp_spa/arp_tpa/arp_sha/arp_tpa without arp)", flow)
+	}
+	if (fieldSet(parsed, "tcp_src") || fieldSet(parsed, "tcp_dst")) && !fieldSet(parsed, "tcp") {
+		return nil, fmt.Errorf("bad flow %q (specified tcp_src/tcp_dst without tcp)", flow)
+	}
+	if (fieldSet(parsed, "udp_src") || fieldSet(parsed, "udp_dst")) && !fieldSet(parsed, "udp") {
+		return nil, fmt.Errorf("bad flow %q (specified udp_src/udp_dst without udp)", flow)
+	}
+	if (fieldSet(parsed, "tp_src") || fieldSet(parsed, "tp_dst")) && !(fieldSet(parsed, "tcp") || fieldSet(parsed, "udp")) {
+		return nil, fmt.Errorf("bad flow %q (specified tp_src/tp_dst without tcp/udp)", flow)
+	}
+	if fieldSet(parsed, "ip_frag") && (fieldSet(parsed, "tcp") || fieldSet(parsed, "udp")) {
+		return nil, fmt.Errorf("bad flow %q (specified ip_frag with tcp/udp)", flow)
+	}
+
+	return parsed, nil
+}
+
+// flowMatches tests if flow matches match. If exact is true, then the table, priority,
+// and all fields much match. If exact is false, then the table and any fields specified
+// in match must match, but priority is not checked, and there can be additional fields
+// in flow that are not in match.
+func flowMatches(flow, match *ovsFlow, exact bool) bool {
+	if flow.table != match.table && (exact || match.table != 0) {
+		return false
+	}
+	if exact && flow.priority != match.priority {
+		return false
+	}
+	if exact && len(flow.fields) != len(match.fields) {
+		return false
+	}
+	if match.cookie != "" && !fieldMatches(flow.cookie, match.cookie, exact) {
+		return false
+	}
+	for _, matchField := range match.fields {
+		var flowValue *string
+		for _, flowField := range flow.fields {
+			if flowField.name == matchField.name {
+				flowValue = &flowField.value
+				break
+			}
+		}
+		if flowValue == nil || !fieldMatches(*flowValue, matchField.value, exact) {
+			return false
+		}
+	}
+	return true
+}
+
+func fieldMatches(val, match string, exact bool) bool {
+	if val == match {
+		return true
+	}
+	if exact {
+		return false
+	}
+	split := strings.Split(match, "/")
+	if len(split) == 2 {
+		matchNum, err1 := strconv.ParseUint(split[0], 0, 32)
+		mask, err2 := strconv.ParseUint(split[1], 0, 32)
+		valNum, err3 := strconv.ParseUint(val, 0, 32)
+		if err1 == nil && err2 == nil && err3 == nil {
+			if (matchNum & mask) == (valNum & mask) {
+				return true
+			}
+		}
+	}
+
+	return false
+}
+
+type ovsFlows []ovsFlow
+
```

Comment:
Comment here is helpful. Len(), Swap(), Less() are implemented to support sort.Interface on ovsFlows